pkg/providers/vsphere/vsphere.go

```diff
@@ -858,6 +860,26 @@ func (p *vsphereProvider) SetupAndValidateCreateCluster(ctx context.Context, clu
 	if err != nil {
 		return fmt.Errorf("failed setup and validations: %v", err)
 	}
+
+	if clusterSpec.IsManaged() {
+		for _, mc := range p.MachineConfigs() {
+			em, err := p.providerKubectlClient.SearchVsphereMachineConfig(ctx, mc.GetName(), clusterSpec.ManagementCluster.KubeconfigFile, mc.GetNamespace())
+			if err != nil {
+				return err
+			}
+			if len(em) > 0 {
+				return fmt.Errorf("VSphereMachineConfig %s already exists", mc.GetName())
```

Comment:
could we check all of the MachineConfigs for duplicate names and then return an error the indicates those MCs which are duplicates, rather than returning an error on the first duplicate we hit? This way multiple duplicate machine configs can be fixed by the user all at once.

Reply:
I'm thinking if we could reuse the `GetEksaVSphereMachineConfig` instead of `Searchxxx` and just add an error message type of `ResourceNotFound`. So that we don't replicate all the Get methods.

Reply:
Also, what do u think abt moving these checks to the preflight validation instead of here?

Reply:
chatted with @mdsgabriel offline, gonna work on it in a separate PR. (https://github.com/aws/eks-anywhere/issues/539)